From 27d43b8304ce47d4b169aa56fdac5a93a36b2f93 Mon Sep 17 00:00:00 2001
From: Alexey Sotkin <alexey.sotkin@intel.com>
Date: Fri, 28 Sep 2018 10:40:21 +0300
Subject: [PATCH 8/9] Fixing lit tests in travis ci.

Building dependencies for debug info tests takes too long,
and Travis CI terminates the build due to time out.
This commit introduces SKIP_SPIRV_DEBUG_INFO_TEST variable in
cmake files. Set to true it disables tests in test/DebugInfo
directory and excludes tools required for these tests from build.
By default the variable is not set, i.e. debug info tests for
SPIR-V will run.
---
 .travis.yml                  |  1 +
 test/CMakeLists.txt          | 22 ++++++++++++++--------
 test/DebugInfo/lit.local.cfg |  6 +++++-
 test/lit.site.cfg.py.in      |  1 +
 4 files changed, 21 insertions(+), 9 deletions(-)

diff --git a/.travis.yml b/.travis.yml
index 3bf7700..dedf829 100644
--- a/.travis.yml
+++ b/.travis.yml
@@ -93,6 +93,7 @@ script:
         -DLLVM_BUILD_TOOLS=ON \
         -DLLVM_BUILD_TESTS=ON \
         -DLLVM_INCLUDE_TESTS=ON \
+        -DSKIP_SPIRV_DEBUG_INFO_TESTS=ON \
         -DLLVM_LIT_ARGS="-sv --no-progress-bar" \
         -G "Unix Makefiles"
     fi
diff --git a/test/CMakeLists.txt b/test/CMakeLists.txt
index b6d2aa7..6c84719 100644
--- a/test/CMakeLists.txt
+++ b/test/CMakeLists.txt
@@ -1,3 +1,5 @@
+llvm_canonicalize_cmake_booleans(SKIP_SPIRV_DEBUG_INFO_TESTS)
+
 # required by lit.site.cfg.py.in
 get_target_property(LLVM_SPIRV_DIR llvm-spirv BINARY_DIR)
 set(LLVM_SPIRV_TEST_SOURCE_DIR ${CMAKE_CURRENT_SOURCE_DIR})
@@ -10,25 +12,29 @@ configure_lit_site_cfg(
 
 if(NOT BUILD_EXTERNAL)
   set(LLVM_SPIRV_TEST_DEPS
-    count
     FileCheck
+    count
+    llvm-as
+    llvm-dis
     not
   )
+  if(NOT SKIP_SPIRV_DEBUG_INFO_TESTS)
+    list(APPEND LLVM_SPIRV_TEST_DEPS
+      llc
+      llvm-dwarfdump
+      llvm-objdump
+      llvm-readobj
+    )
+  endif(NOT SKIP_SPIRV_DEBUG_INFO_TESTS)
 endif(NOT BUILD_EXTERNAL)
 
+
 add_lit_testsuite(check-llvm-spirv "Running the LLVM-SPIRV regression tests"
   ${CMAKE_CURRENT_BINARY_DIR}
   ARGS
     --verbose
   DEPENDS
     ${LLVM_SPIRV_TEST_DEPS}
-    llc
-    llvm-as
-    llvm-dis
-    llvm-dwarfdump
-    llvm-objdump
-    llvm-readelf
-    llvm-readobj
     llvm-spirv
 )
 
diff --git a/test/DebugInfo/lit.local.cfg b/test/DebugInfo/lit.local.cfg
index eb016c7..36dbb27 100644
--- a/test/DebugInfo/lit.local.cfg
+++ b/test/DebugInfo/lit.local.cfg
@@ -1,2 +1,6 @@
-config.suffixes = ['.ll']
 config.substitutions.append(('%triple', config.target_triple))
+
+# Building dependencies for debug info tests takes too long,
+# so travis ci terminates the build after time out.
+if config.skip_spirv_debug_info_tests:
+    config.unsupported = True
diff --git a/test/lit.site.cfg.py.in b/test/lit.site.cfg.py.in
index 61c6684..2dfd52a 100644
--- a/test/lit.site.cfg.py.in
+++ b/test/lit.site.cfg.py.in
@@ -15,6 +15,7 @@ config.target_triple = "@TARGET_TRIPLE@"
 config.host_arch = "@HOST_ARCH@"
 config.python_executable = "@PYTHON_EXECUTABLE@"
 config.test_run_dir = "@CMAKE_CURRENT_BINARY_DIR@"
+config.skip_spirv_debug_info_tests = @SKIP_SPIRV_DEBUG_INFO_TESTS@
 
 # Support substitution of the tools and libs dirs with user parameters. This is
 # used when we can't determine the tool dir at configuration time.
-- 
1.8.3.1

